sphinx/texinputs/sphinx.sty

```diff
@@ -1881,9 +1881,10 @@
     \let\sphinxtermref  \@firstofone
 }}
 
-% For curly braces inside \index macro
-\def\sphinxleftcurlybrace{\{}
-\def\sphinxrightcurlybrace{\}}
+% For curly braces and vertical bar inside \index macro
+\def\sphinxleftcurlybrace#1{\{}
+\def\sphinxrightcurlybrace#1{\}}
+\def\sphinxverticalbar#1{|}
```

Comment:
The extra `#1` serve to swallow the brace pairs `{}` added after the macro names (mandatory to avoid letters following glued to macro name) thus producing better `.idx` files with no extraneous `{}`'s.